modules/service_role/main.tf

```diff
@@ -0,0 +1,36 @@
+resource "aws_iam_role" "main" {
+  name               = var.role_name
+  description        = var.role_description
+  assume_role_policy = data.aws_iam_policy_document.services_can_assume.json
+  tags               = var.tags
+}
+
+data "aws_iam_policy_document" "services_can_assume" {
+  statement {
+    sid     = "AssumeRole"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "service"
+      identifiers = var.services
+    }
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "main" {
+  for_each = merge(
+    var.policies,
+    zipmap(var.policy_documents, aws_iam_policy.main[*].arn)
+  )
```

Comment:
I'm always very cautious about this because of the hard limit (10) on policy attachments per role. A few options to consider:
a) validate that the combined length of `var.policies` and `var.policy_documents` is `<= 10`
b) collapse all policy documents into a single policy via [source_policy_documents](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document#source_policy_documents)
c) leave policy attachments out of this module entirely as the client's responsibility

Note for (a): you can't write "traditional" variable validations that reference other vars (you can only refer to "yourself" within the context of a `variable`) but we've done some multi-var validations by doing something like this:
```
locals {
  total_policies_passed_in = len(var.policies) + len(var.policy_documents)
  cannot_attach_more_than_ten_policies = coalesce(local.total_policies_passed_in <= 10 ? "ok" : null)
}
```
`coalesce` will blow up at plan time if it does not resolve to a non-null value.

Reply:
Oh, good point, I'll put in that validation logic.